android/julius/src/main/java/bvschaik/julius/JuliusMainActivity.java

```diff
@@ -0,0 +1,120 @@
+package bvschaik.julius;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.widget.Toast;
+import org.libsdl.app.SDLActivity;
+
+public class JuliusMainActivity extends SDLActivity
+{
+    private static final int GET_FOLDER_RESULT = 500;
+    private static final int rwFlagsPermission = Intent.FLAG_GRANT_READ_URI_PERMISSION
+                                               | Intent.FLAG_GRANT_WRITE_URI_PERMISSION;
+    public static boolean paused = true;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public void onStart()
+    {
+        super.onStart();
+        paused = false;
+    }
+
+    @Override
+    public void onStop()
+    {
+        super.onStop();
+        FileManager.folderStructureCache.clear();
+    }
+
+    @Override
+    protected String[] getLibraries() {
+        return new String[]{
+            "SDL2",
+            "SDL2_mixer",
+            "julius"
+        };
+    }
+
+    @Override
+    protected void onPause()
+    {
+        super.onPause();
+        paused = true;
+    }
+
+    @Override
+    protected void onResume()
+    {
+        super.onResume();
+        paused = false;
+    }
+
+    public void showDirectorySelection()
+    {
+        // Wait before showing window
+        Sleep(1000);
```

Comment:
Sleeping on the main thread? Bad idea. Android might kill the app because of an [ANR](https://developer.android.com/topic/performance/vitals/anr) (app not responding).

Reply:
I just wanted to show a toast message before the "select folder" appears, as that makes the toast message disappear before the user has the time to read. Do you know of an alternative? I really don't want to show a message box...